OMCompiler/Compiler/FrontEnd/InstVar.mo

```diff
@@ -1111,66 +1114,73 @@ algorithm
         ci_state = ClassInf.start(res, Absyn.IDENT(cls_name));
         predims = List.lastListOrEmpty(inInstDims);
         pre = PrefixUtil.prefixAdd(inName, predims, idxs, inPrefix, vt, ci_state, inInfo);
-        (cache, env_1, ih, store, dae1, csets, ty,_, opt_attr, graph) =
+        if Config.acceptMetaModelicaGrammar() then
+          stateName = AbsynUtil.pathString(ClassInf.getStateName(inState), delimiter = "");
+          inStateAndClassNameIsEqual = stringEqual(stateName, cls_name);
+          implicitInstantiation = SCodeUtil.isUniontype(inClass)
```

Comment:
Basically we check if a uniontype constant is directly self-recursive. If that is the I create a temporary interface which I do not cache and then proceed as before